cedar-policy/src/api/id.rs

```diff
@@ -160,6 +155,29 @@ impl std::fmt::Display for EntityTypeName {
     }
 }
 
+#[doc(hidden)]
+impl From<ast::Name> for EntityTypeName {
+    fn from(value: ast::Name) -> Self {
+        Self(value)
+    }
+}
+
+#[doc(hidden)]
+impl TryFrom<ast::EntityType> for EntityTypeName {
+    type Error = UnspecifiedCannotBeConvertedToEntityTypeName;
+    fn try_from(value: ast::EntityType) -> Result<Self, Self::Error> {
+        match value {
+            ast::EntityType::Specified(name) => Ok(name.into()),
+            ast::EntityType::Unspecified => Err(UnspecifiedCannotBeConvertedToEntityTypeName),
+        }
+    }
+}
+
+#[doc(hidden)]
+#[derive(Debug, Error)]
+#[error("`Unspecified` cannot be converted to `EntityTypeName`")]
+pub struct UnspecifiedCannotBeConvertedToEntityTypeName;
```

Comment:
Makes sense to wait to merge this until after #983 since that PR make the unspecified error irrelevant.

Reply:
Yes, I will wait.  It also changes the contained type from `Name` to `EntityType` which breaks a bunch of the changes in this PR.

Reply:
Updated